File: src/main/java/com/d20charactersheet/framework/dac/dao/FeatDao.java
package com.d20charactersheet.framework.dac.dao;

import java.util.List;

import com.d20charactersheet.framework.boc.model.Feat;

/**
 * Interface to provide access to persitent feat data.
 */
public interface FeatDao {

  /**
   * Returns all feats.
   *
   * @return All feats.
   */
  List<Feat> getAllFeats();

  /**
   * Create the feat.
   *
   * @param feat
   *     The feat to create.
   */
  void createFeat(Feat feat);

  /**
   * Updates the feat.
   *
   * @param feat
   *     The feat to update.
   */
  void updateFeat(Feat feat);

  /**
   * Deletes the feat.
   *
   * @param feat
   *     The feat to delete.
   */
  void deleteFeat(Feat feat);

}

File: src/main/kotlin/com/d20charactersheet/framework/dac/dao/FeatDao.kt
package com.d20charactersheet.framework.dac.dao

import com.d20charactersheet.framework.boc.model.Feat

/**
 * Interface to provide access to persistent feat data.
 */
interface FeatDao {
    /**
     * Returns all feats.
     *
     * @return All feats.
     */
    fun getAllFeats(): List<Feat>

    /**
     * Create the feat.
     *
     * @param feat
     * The feat to create.
     */
    fun createFeat(feat: Feat)

    /**
     * Updates the feat.
     *
     * @param feat
     * The feat to update.
     */
    fun updateFeat(feat: Feat)

    /**
     * Deletes the feat.
     *
     * @param feat
     * The feat to delete.
     */
    fun deleteFeat(feat: Feat)
}